instrumentation/concurrent_ruby/lib/opentelemetry/instrumentation/concurrent_ruby/instrumentation.rb

```diff
@@ -16,17 +16,21 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         present do
-          defined?(::Concurrent::Future)
+          defined?(::Concurrent::RubyExecutorService) || defined?(::Concurrent::JavaExecutorService)
         end
 
         private
 
         def require_dependencies
-          require_relative 'patches/future'
+          require_relative 'patches/executor_service'
         end
 
         def patch
-          ::Concurrent::Future.send(:include, Patches::Future)
+          if ::Concurrent.on_jruby?
```

Comment:
Could you help me understand this decision here? 
Would it work if this was mixed into `ThreadPoolExectutor` instead?

Reply:
<img width="569" src="https://user-images.githubusercontent.com/51594/133780388-a0083d84-fc9e-4137-a58b-26edefd2c15a.png">

Because, some classes inherit from `RubyExecutorService`. So I mixed into that class.
(But, I don't know Concurrent::TimerTask works at JRuby environment...)

Reply:
Thank you for taking the time to explain that. 

Looking at other API docs it looks like there are a few other Ruby specific implementations so this does indeed seem necessary.
- https://ruby-concurrency.github.io/concurrent-ruby/master/Concurrent/SimpleExecutorService.html 
- https://ruby-concurrency.github.io/concurrent-ruby/master/Concurrent/TimerSet.html

Reply:
Thank you for review. I'll verify that class works at JRuby environment later.

Reply:
🤔 @ericmustin If this is prepended only on the platform specific executor services, then the derived classes may not be instrumented the way we expect. I think I am a little confused by some of the implementations in the derived classes.

`TimerSet` overrides `post` and does not invoke `super` so it will not be auto-instrumented as a result, but that may be ok?

https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent-ruby/concurrent/executor/timer_set.rb#L48

While the `TimerTask` _changes_ the interface by encapsulating the `post` method altogether and introduces a new interface named `execute`, which will also not be auto-instrumented: 

https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent-ruby/concurrent/timer_task.rb#L273

`SimpleExecutorService` _also_ overrides `post` and does not invoke `super`:
https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent-ruby/concurrent/executor/simple_executor_service.rb#L37

❓ Do we want to instrument any of these derived classes? If so then I _think_ it would be outside of the scope of this PR, but that goes back to me wanting this PR to patch the non-platform specific `ThreadPoolExectutor` and then look more closely at the other Ruby platform specific classes to see what/how to auto-instrument those in a future PR.

Reply:
@arielvalentin 

Hi, I tried below codes MRI and JRuby. But, it couldnot instrument as expected. because `Concurrent::ScheduledTask` does not use concurrent executor service.

```ruby
tracer.in_span('outer_span') do
  task = Concurrent::ScheduledTask.execute(0.5) do
    p Thread.current.object_id
    tracer.in_span('inner_span') {
      p Thread.current.object_id
      puts "Concurrent::ScheduledTask"
    }
  end
  p task.state
end
sleep 1 while task.pending?
p task.state

exporter.finished_spans.each_with_index do |span, i|
  puts "i=#{i} name=#{span.name} span_id=#{span.span_id.inspect} parent_span_id=#{span.parent_span_id.inspect}"
end
```

> wanting this PR to patch the non-platform specific ThreadPoolExectutor and then look more 

So I changing my mind. agree with your option.